astroplan/core.py

```diff
@@ -162,171 +207,851 @@ def altaz(self, time, target=None, obswl=None):
+            like to calculate. Default is nearest.
+
+        Returns
+        -------
+        `~astropy.time.Time`
+            Time of sunset
         """
-        raise NotImplementedError()
+        return self.sunrise(time, which, horizon=-6*u.degree)
+
+    # Moon-related methods.
 
-    def evening_civil(self, time):
+    def moonrise(self, time, **kwargs):
         """
-        Returns the local evening (civil) twilight time.
+        Returns the local moonrise time.
+
+        The default moonrise returned is the next one to occur.
 
         Parameters
         ----------
-        time : `~astropy.time.Time`
+        time : `~astropy.time.Time` or other (see below)
+
+        Keywords: str, optional
+            previous
+            next
         """
         raise NotImplementedError()
 
-    def morning_astronomical(self, time):
+    def moonset(self, time, **kwargs):
         """
-        Returns the local morning (astronomical) twilight time.
+        Returns the local moonset time.
+
+        The default moonset returned is the next one to occur.
 
         Parameters
         ----------
-        time : `~astropy.time.Time`
+        time : `~astropy.time.Time` or other (see below)
+            This will be passed in as the first argument to
+            the `~astropy.time.Time` initializer, so it can be anything that
+            `~astropy.time.Time` will accept (including a `~astropy.time.Time`
+            object).
+
+        Keywords: str, optional
+            previous
+            next
         """
         raise NotImplementedError()
 
-    def morning_nautical(self, time):
+    def moon_illumination(self, time):
         """
-        Returns the local morning (nautical) twilight time.
+        Returns a float giving the percent illumation.
 
         Parameters
         ----------
-        time : `~astropy.time.Time`
+        time : `~astropy.time.Time` or other (see below)
+            This will be passed in as the first argument to
+            the `~astropy.time.Time` initializer, so it can be anything that
+            `~astropy.time.Time` will accept (including a `~astropy.time.Time`
+            object).
         """
         raise NotImplementedError()
 
-    def morning_civil(self, time):
+    def moon_position(self, time):
         """
-        Returns the local morning (civil) twilight time.
+        Returns the position of the moon in alt/az.
 
         Parameters
         ----------
-        time : `~astropy.time.Time`
+        time : `~astropy.time.Time` or other (see below)
+            This will be passed in as the first argument to
+            the `~astropy.time.Time` initializer, so it can be anything that
+            `~astropy.time.Time` will accept (including a `~astropy.time.Time`
+            object).
         """
         raise NotImplementedError()
 
+    @u.quantity_input(horizon=u.deg)
+    def can_see(self, time, target, horizon=0*u.degree, ret_altaz=False):
```

Comment:
One reason we really preferred `can_see` is because it's a method on the Observer class, so you could read it as `keck.can_see(Time.now(), vega)`. Otherwise you'd be calling `is_visible` on the observatory.